Identified class of popup

reach2jeyan opened the pull request:
@divhegde 

I have tried identifying the class name as `popupcontent` which is actually identifying the contents inside the drop down box. But here, there is one more catch, I cannot identify the text inside the above div class though it exists, since, these texts are with the different class names.

I tried, to make the `popupcontent` as parent class, and then add child class(which is associated to text) and then identifying that childclass with the text name, but this idiot, will still not select it.

Could you try, from here, how to select the content with the above class name `popupcontent`.

I have added the step to print the contents of the drop down box to verify whether this is the correct parent class. Also, To identify the class of each content, You can try printing the class names that exists with these texts inside the drop down box and check? 

Let me know

emjeyforta committed:
Identified class of popup

divhegde commented on features/step_definitions/test_steps.rb:
```diff
@@ -204,7 +204,29 @@
+Then(/^I select "([^"]*)" with index "([^"]*)" with div class "([^"]*)"$/) do |name, index, classname|
+
+end
+
```
If no content, can we  remove the step definition

reach2jeyan replied:
Sure! Will remove the empty step definitions in the next commit.

divhegde commented on features/step_definitions/test_steps.rb:
```diff
@@ -204,7 +204,29 @@
+    puts div.text
+  end
+end
+
```
Can we merge this one with the step def for `Then I print the prices` ?  Both have same commands, only different statements

reach2jeyan replied:
Agreed, Both are same! but, this step, I introduced it, to only verify, as to what `class` is holding those contents in the drop box, so that the valid class would print the contents. Now that, the class is identified, once the element in the class is able to be clicked, I ll remove this step, and proceed to modifying step definitions in the scenarios you have written. Hence, not merging this.

reach2jeyan replied:
```
Then(/^I print the text with the class name "([^"]*)"$/) do |classname|
  text = browser.div(:class => classname)
  puts text.text
```

Used this to identify the class names of the induvidual options in the drop down box.

Looks like all the class names below, just returns the text `Economy class`

![image](https://cloud.githubusercontent.com/assets/20741047/23114458/3b89b4c8-f765-11e6-8ad8-60f00a01b18d.png)

![image](https://cloud.githubusercontent.com/assets/20741047/23114491/5c820df6-f765-11e6-9875-435da9afff79.png)

![image](https://cloud.githubusercontent.com/assets/20741047/23114505/6d2aafdc-f765-11e6-8620-abacee456403.png)

reach2jeyan replied:
And Hence, I am proceeding towards, clicking the element using `XPATH` and this has fixed the issue.

![image](https://cloud.githubusercontent.com/assets/20741047/23114665/668e94c6-f766-11e6-9e61-5bdc23361da5.png)

divhegde commented on features/step_definitions/test_steps.rb:
```diff
@@ -204,7 +204,29 @@
+
+Then(/^I select in the dropdown box content with class "([^"]*)" select option "([^"]*)" with div class "([^"]*)"$/) do |parentclass, name, childclass|
+  browser.select_list(:class => parentclass).div(:class => childclass).set(:text => name)
 end
```
Doesnt `set(:text => name)` have same function as send_keys?

reach2jeyan replied:
Theoretically Yes!

emjeyforta committed:
Bug fixes

emjeyforta committed:
Bug fixes

reach2jeyan commented:
Changelogs:

1. Fixed flight scenario by impelementing `xpath` to identify the element. Modified existing flight scenario steps, since one element associated to multiple classnames and viceversa. Also, Since, multiple class names only return same element.

```
Then(/^I select "([^"]*)" in the drop down box with xpath "([^"]*)"$/) do |name, path|
  browser.element(:xpath => path).click
end
```

2. Removed all the unnecessary and empty codes in step_definition file
3. Fixed unrelated PR failure, wrt Amazon purchase, failing at adding item to the cart, since the item, mentioned in the test step, not displayed in the page. Added new item now.
4. Cucumber html report enhancement feature, removed in the latest commit.

@divhegde Check it out. Also, I guess the scenario with respect to adding number of child and adults, `+` and `-` need to be identified using `xpath` itself, since multiple tabs exists with same or multiple class names and do not have an unique identifier.

You can work on that and push commits to this PR itself or merge it and automate the above?
